src/physics.F90

```diff
@@ -1182,35 +1165,41 @@ subroutine create_fission_sites(p, i_nuclide, i_reaction, bank_array, size_bank)
   end subroutine create_fission_sites
 
 !===============================================================================
-! SAMPLE_FISSION_ENERGY
+! SAMPLE_FISSION_NEUTRON
 !===============================================================================
 
-  function sample_fission_energy(nuc, rxn, p) result(E_out)
-
-    type(NuclideCE), intent(in) :: nuc
-    type(Reaction),   intent(in) :: rxn
-    type(Particle), intent(inout) :: p     ! Particle causing fission
-    real(8)                       :: E_out ! outgoing energy of fission neutron
+  subroutine sample_fission_neutron(nuc, rxn, E_in, site)
```

Comment:
Not a bad idea; I'd like to rethink all of the routines in the physics module. Maybe as a separate PR though? Are you ok deferring?